src/main/java/com/puppycrawl/tools/checkstyle/gui/TreeTable.java

```diff
@@ -193,62 +195,74 @@ private void setColumnsInitialWidth() {
         getColumn("Type").setPreferredWidth(preferredTypeColumnWidth);
     }
 
-    /**
-     * Search node by Xpath.
-     *
-     * @param root {@code DetailAST} root ast element
-     * @param xpath {@code String} xpath query
-     * @param nodes {@code Deque<DetailAST>} stack of nodes in selection path
-     * @return {@code boolean} node found or not
-     */
-    private static boolean search(DetailAST root, String xpath, Deque<DetailAST> nodes) {
-        boolean result = false;
-        if (xpath.equals(XpathQueryGenerator.generateXpathQuery(root))) {
-            nodes.push(root);
-            result = true;
-        }
-        else {
-            DetailAST child = root.getFirstChild();
-            while (child != null) {
-                if (search(child, xpath, nodes)) {
-                    nodes.push(root);
-                    result = true;
-                    break;
-                }
-                child = child.getNextSibling();
-            }
-        }
-        return result;
-    }
-
     /**
      * Select Node by Xpath.
      */
     public void selectNodeByXpath() {
         final DetailAST rootAST = (DetailAST) tree.getModel().getRoot();
         if (rootAST.hasChildren()) {
-            final String xpath = "/EOF" + xpathEditor.getText();
+            final String xpath = xpathEditor.getText();
             final Deque<DetailAST> nodes = new ArrayDeque<>();
-            if (search(rootAST, xpath, nodes)) {
-                TreePath path = new TreePath(nodes.pop());
-                while (!nodes.isEmpty()) {
-                    path = path.pathByAddingChild(nodes.pop());
-                    if (!tree.isExpanded(path)) {
-                        tree.expandPath(path);
-                    }
-                    tree.setSelectionPath(path);
-                    makeCodeSelection();
-                }
+
+            try {
+                final List<DetailAST> xPathItems =
+                        XpathUtil.getXpathItems(xpath, new RootNode(rootAST))
+                                .stream()
+                                .map(item -> ((AbstractNode) item).getUnderlyingNode())
+                                .collect(Collectors.toList());
+                nodes.addAll(xPathItems);
```

Comment:
there is no need for intermediate list, you can just finish stream with `.forEach(nodes::add)`

Reply:
Or, it is possible to collect right into the deque:
```java
nodes = XpathUtil.getXpathItems(xpath, new RootNode(rootAST))
    .stream()
    .map(item -> ((AbstractNode) item).getUnderlyingNode())
    .collect(Collectors.toCollection(ArrayDeque<DetailAST>::new));
```

Reply:
Done, used method reference for cast, too.